nemo_curator/utils/file_utils.py

```diff
@@ -195,11 +194,22 @@ def _name(start_index, npad, prefix, i):
 
     output_glob_string = os.path.join(output_path, "*.jsonl")
 
-    documents.map(_output_json).to_textfiles(
+    output_files = documents.map(_output_json).to_textfiles(
         output_glob_string,
         name_function=name,
     )
 
+    # Delete empty files generated due to empty partitions in the bag
+    for output_file in output_files:
+        try:
+            if os.path.getsize(output_file) == 0:
+                os.remove(output_file)
+        except Exception as exception:
+            print(
+                f"An exception occurred when trying to delete {output_file}.\n{exception}",
+                flush=True,
+            )
+
 
 def reshard_jsonl(
     input_dir, output_dir, output_file_size="100M", start_index=0, file_prefix=""
```

Comment:
If you could change the docstring for `start_index` on line 225 to add something like "Note: The indices may not be continuous if the sharding process would output an empty file in its place" (I would add this comment on line 225 itself, but GitHub isn't letting me right now for some reason)

Reply:
Hi @ryantwolf , done!

Thanks!